index.js

```diff
@@ -1122,7 +1141,15 @@
         if (result != null && result !== '[object Object]') return result;
       }
 
-      return call(implPath([xType, 'prototype', 'toString']) || x.toString, x);
+      var f = implementations[xType] != null &&
+              has('toString', implementations[xType].prototype) ?
+              implementations[xType].prototype.toString :
+              typeof x.toString === 'function' ?
+              x.toString :
+              Object.prototype.toString;
```

Comment:
This is here for two reasons:

1. We fix:

```js
> var Bad = function(){}
> Bad['@@type'] = 'NotObject'
> Bad.prototype.toString = 'nonsense'

> Z.toString(new Bad)
TypeError: method.call is not a function
```

Now this code evaluates to `[object Object]`.

2. `methodImpl` no longer needs to check for `hasOwnProperty`. The only culprit was `toString`.